pkg/cluster/loadbalancerprofile.go

```diff
@@ -0,0 +1,386 @@
+	if m.doc.OpenShiftCluster.Properties.NetworkProfile.OutboundType != api.OutboundTypeLoadbalancer {
+		return nil
+	}
+
+	resourceGroupName := stringutils.LastTokenByte(m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID, '/')
+	infraID := m.doc.OpenShiftCluster.Properties.InfraID
+
+	lb, err := m.loadBalancers.Get(ctx, resourceGroupName, infraID, "")
+	if err != nil {
+		return err
+	}
+
+	err = m.reconcileOBRuleV4OBIPs(ctx, lb)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Reconcile the outbound rule "outbound-rule-v4" frontend IP Config.
+func (m *manager) reconcileOBRuleV4OBIPs(ctx context.Context, lb mgmtnetwork.LoadBalancer) error {
+	m.log.Info("reconciling outbound-rule-v4")
+	defer func() {
+		err := m.deleteUnusedManagedIPs(ctx)
+		if err != nil {
+			m.log.Error("failed to cleanup unused managed IPs, error: %w", err)
+		}
+	}()
+
+	resourceGroupName := stringutils.LastTokenByte(m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID, '/')
+	infraID := m.doc.OpenShiftCluster.Properties.InfraID
+	originalOutboundIPs := getOutboundIPsFromLB(lb)
+	// ensure effectiveOutboundIPs is patched the first time running against a cluster
+	if m.doc.OpenShiftCluster.Properties.NetworkProfile.LoadBalancerProfile.EffectiveOutboundIPs == nil {
+		err := m.patchEffectiveOutboundIPs(ctx, originalOutboundIPs)
+		if err != nil {
+			return err
+		}
+	}
+
+	desiredOutboundIPs, err := m.getDesiredOutboundIPs(ctx)
+	if err != nil {
+		return err
+	}
+
+	if areResourceRefsEqual(desiredOutboundIPs, originalOutboundIPs) {
+		return nil
+	}
+
+	// rebuild outbound-rule-v4 frontend ip config with desired outbound ips
+	removeOutboundIPsFromLB(lb)
+	addOutboundIPsToLB(m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID, lb, desiredOutboundIPs)
+
+	err = m.loadBalancers.CreateOrUpdateAndWait(ctx, resourceGroupName, infraID, lb)
+	if err != nil {
+		return err
+	}
+
+	// update database with new effective outbound IPs
+	err = m.patchEffectiveOutboundIPs(ctx, desiredOutboundIPs)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Remove all frontend ip config in use by outbound-rule-v4.  Frontend IP config that is used by load balancer rules will be saved.
+func removeOutboundIPsFromLB(lb mgmtnetwork.LoadBalancer) {
+	// get all outbound rule fip config to remove
+	var obRuleV4FIPConfigMap = make(map[string]mgmtnetwork.SubResource)
+	for _, obRule := range *lb.LoadBalancerPropertiesFormat.OutboundRules {
+		if *obRule.Name == "outbound-rule-v4" {
+			for i := 0; i < len(*obRule.OutboundRulePropertiesFormat.FrontendIPConfigurations); i++ {
+				fipConfigID := *(*obRule.OutboundRulePropertiesFormat.FrontendIPConfigurations)[i].ID
+				fipConfig := (*obRule.OutboundRulePropertiesFormat.FrontendIPConfigurations)[i]
+				obRuleV4FIPConfigMap[fipConfigID] = fipConfig
+			}
+			// clear outbound-rule-v4 frontend ip config
+			*obRule.FrontendIPConfigurations = []mgmtnetwork.SubResource{}
```

Comment:
nit: can optimize with a `break` here, if we're only expecting one Rule with this `Name`.